Tests/Unit/NetworkServer/LnsProtocolMessageProcessorTests.cs

```diff
@@ -284,29 +283,19 @@ internal async Task InternalHandleDataAsync_ShouldNotThrow_OnNonHandledMessageTy
                                                                        CancellationToken.None);
         }
 
-        private static Rxpk GetExpectedRxpk()
+        private static RadioMetadata GetExpectedRadioMetadata()
         {
             var radioMetadataUpInfo = new RadioMetadataUpInfo(0, 68116944405337035, 0, -53, (float)8.25);
-            var radioMetadata = new RadioMetadata(DataRateIndex.DR5, new Hertz(868300000), radioMetadataUpInfo);
-            return new BasicStationToRxpk(radioMetadata, RegionManager.EU868);
+            return new RadioMetadata(DataRateIndex.DR5, new Hertz(868300000), radioMetadataUpInfo);
         }
 
-        private static bool AreRxpkEqual(Rxpk subject, Rxpk other) =>
-            subject.Chan == other.Chan
-            && subject.Codr == other.Codr
-            && subject.Data == other.Data
-            && subject.Datr == other.Datr
-            && subject.Freq == other.Freq
-            && subject.Lsnr == other.Lsnr
-            && subject.Modu == other.Modu
-            && subject.RequiredSnr == other.RequiredSnr
-            && subject.Rfch == other.Rfch
-            && subject.Rssi == other.Rssi
-            && subject.Size == other.Size
-            && subject.Stat == other.Stat
-            && subject.Time == other.Time
-            && subject.Tmms == other.Tmms
-            && subject.Tmst == other.Tmst;
+        private static bool RadioMetadata(RadioMetadata subject, RadioMetadata other) =>
+            subject.DataRate == other.DataRate
+            && subject.Frequency == other.Frequency
+            && subject.UpInfo.Xtime == other.UpInfo.Xtime
+            && subject.UpInfo.SignalNoiseRatio == other.UpInfo.SignalNoiseRatio
+            && subject.UpInfo.GpsTime == other.UpInfo.GpsTime
+            && subject.UpInfo.ReceivedSignalStrengthIndication == other.UpInfo.ReceivedSignalStrengthIndication;
 
```

Comment:
Since you made `RadioMetadata` a record, you get equality comparison for free so you can completely remove this helper:

```suggestion
```

See and commit together with [my other suggestion](https://github.com/Azure/iotedge-lorawan-starterkit/pull/1122/files#r772525778).